src/pages/instances/CreateInstance.tsx

```diff
@@ -314,19 +314,19 @@ const CreateInstance: FC = () => {
     }
     void formik.setFieldValue("devices", devices);
 
-    if (isVmOnlyImage(image)) {
+    if (type) {
+      void formik.setFieldValue("instanceType", type);
+    } else if (isVmOnlyImage(image)) {
       void formik.setFieldValue("instanceType", "virtual-machine");
     } else if (isContainerOnlyImage(image)) {
       void formik.setFieldValue("instanceType", "container");
-    } else if (type) {
-      void formik.setFieldValue("instanceType", type);
     }
   };
 
   useEffect(() => {
     if (location.state?.selectedImage) {
       const type = location.state.selectedImage.volume
-        ? "iso-volume"
+        ? "virtual-machine"
```

Comment:
maybe here we can benefit from a comment stating why they type is set to `virtual-machine` if `volume` exists on `selectedImage`?

Reply:
I refactored this a bit so it becomes more straight forward, and we don't need the branch here.